### PR TITLE
Handle transformer renaming columns not breaking destination data types in integration pipelines

### PR DESCRIPTION
**Pull request by splatcollision**
In case of integration pipeline transformer renaming columns, support an optional 'aliases' field in the schema properties, that maps source column data types to the renamed column data types.

Slack thread: https://mageai.slack.com/archives/C03HTTWFEKE/p1694214865146399

# Description

Since data passed to transform blocks is a pandas DataFrame, many operations are available on each row of your data.

It's possible to do a few dangerous things with this, for example if you have a target destination that needs it's columns renamed to match existing schemas.

In this case, there can be edge case mismatches between the data types in the source columns, and the expected data types in the destination. This is particularly important for SQL columns where the INSERT commands generated by the pipelines will use CAST expressions to try and make sure the writes succeed. However, if a column is renamed in the transformer, we will use our best guess by looking at values to decide what to CAST the column values to.

For example, a TIMESTAMPTZ value of `2023-01-09T20:01:00`, once it's renamed, can look like a string, but using an INSERT with `CAST('2023-01-09T20:01:00' AS TEXT)` to a column with data type `TIMESTAMPTZ` will fail.

In order to fix this, you can add column "aliases" in the pipeline's data integration catalog.

(Examples of transformer block and catalog json changes in doc changes within this PR)

# How Has This Been Tested?

1. I've encountered insert problems when renaming dataframe columns in my transformer blocks. In particular, TIMESTAMPTZ columns attempt to insert as TEXT via the CAST statements.
2. I added 'aliases' property to each schema property in the data integration catalog file.
3. I update the code on my local mage with the change in this PR.
4. I added extra pipeline logging to confirm the alias maps were being created correctly (now removed)
5. I checked the pipeline editing UI, and making changes there does not affect the custom changes in the catalog file.
6. Without having to provide default CASTs in the postgres layer, the integration pipeline INSERTs now CAST as the correct data types, and the inserts succeed without SQL errors.

# Checklist
- [ ] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

cc:
@wangxiaoyou1993 @tommydangerous

### CODE REVIEW

File: docs/design/blocks/transformer.mdx [hunk 1]
```diff
@@ -44,3 +44,120 @@ import pandas as pd
 def transform_df(df1, df2, df3, **kwargs) -> DataFrame:
     return pd.concat([df1, df2, df3])
 ```
+
+## Handling column renames
```
Comment (wangxiaoyou1993) [inline docs/design/blocks/transformer.mdx:48]: this section is for data integration pipeline. could you update the section name to "Handling column renames in data integration pipeline"

File: mage_ai/data_preparation/models/block/integration/__init__.py [hunk 2]
```diff
@@ -246,6 +254,14 @@ def __update_col_names(schema_dict: Dict, properties_key: str, new_columns: List
                                         if k in properties_original else v
                                         for k, v in properties_updated.items()
                                     }
+                                    # Separately, update schema types from alias map
+                                    if len(output_col_aliases.keys()):
+                                        for cname in schema_updated['schema']['properties']:
+                                            if cname in output_col_aliases:
+                                                prop = output_col_aliases[cname]
+                                                schema_updated['schema']['properties'][cname] = prop
+
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/block/integration/__init__.py:263]: nit: remove extra blank line according to the code quality check (flake8)

File: mage_ai/data_preparation/models/block/integration/__init__.py [hunk 3]
```diff
@@ -246,6 +254,14 @@ def __update_col_names(schema_dict: Dict, properties_key: str, new_columns: List
                                         if k in properties_original else v
                                         for k, v in properties_updated.items()
                                     }
+                                    # Separately, update schema types from alias map
+                                    if len(output_col_aliases.keys()):
+                                        for cname in schema_updated['schema']['properties']:
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/block/integration/__init__.py:259]: I think it should be `for cname in schema_updated['schema']['properties'].keys()`

Comment (splatcollision) [reply mage_ai/data_preparation/models/block/integration/__init__.py]: Sure, updated

File: mage_ai/data_preparation/models/block/integration/__init__.py (hunk 2 above)
Comment (splatcollision) [reply mage_ai/data_preparation/models/block/integration/__init__.py]: 👍

File: docs/design/blocks/transformer.mdx (hunk 1 above)
Comment (splatcollision) [reply docs/design/blocks/transformer.mdx]: Absolutely

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**